contrib/epee/src/readline_buffer.cpp

```diff
@@ -196,15 +200,50 @@ static int startup_hook()
   return 0;
 }
 
+static char* completion_matches(const char* text, int state)
+{
+  static size_t list_index;
+  static size_t len;
+
+  if(state == 0)
+  {
+    list_index = 0;
+    len = strlen(text);
+  }
+
+  const std::vector<std::string>& completions = rdln::readline_buffer::get_completions();
+  for(; list_index<completions.size(); )
+  {
+    const std::string& cmd = completions[list_index++];
+    if(cmd.compare(0, len, text) == 0)
```

Comment:
Wait, spoke too soon. This is a left-bounded search, whereas find is unanchored. So less ops in this version in many cases. Really wanted to drop that darned static variable.